Dynatrace/dynatrace-operator
# Move OneAgent delete logic into oneagent_reconciler

## Description

Just a smaller refactor, so we reduce the responsibility of the `dynakube-controller`
 
## How can this be tested?

No change compared to current behaviour.

If you deploy a Dynakube with a OneAgent, then turn it off, then the daemonset will be removed.
If you deploy a Dynakube without a OneAgent, nothing should break.

## Checklist
